sound/soc/sof/pcm.c

```diff
@@ -363,6 +366,12 @@ static int sof_pcm_trigger(struct snd_pcm_substream *substream, int cmd)
 		stream.hdr.cmd |= SOF_IPC_STREAM_TRIG_START;
 		break;
 	case SNDRV_PCM_TRIGGER_SUSPEND:
+		/* Don't suspend the d0i3 compatible stream */
```

Comment:
well it depends on the target state. If you target S0, then you should not suspend. If you target S3, then yes you should absolutely suspend.

When I made a reference in earlier reviews that we cannot have explicit acpi references, it doesn't mean we need to forget about the concept. I was really asking for an abstraction that compiles when CONFIG_ACPI is not selected.

Reply:
I got your point, I intended to make logic simpler with the assumption that when d0i3 stream opened, we will only enter d0i3 and never do power off to the ADSP. The cons of this is that we can't handle S3 suspend for audio part.

Okay, let me keep the check with fixing the compiling issue when CONFIG_ACPI not selected.